mypy/checker.py

```diff
@@ -2785,6 +2788,7 @@ def infer_variable_type(self, name: Var, lvalue: Lvalue,
         init_type = get_proper_type(init_type)
         if isinstance(init_type, DeletedType):
             self.msg.deleted_as_rvalue(init_type, context)
+            name.type = AnyType(TypeOfAny.from_error)
```

Comment:
Is there a test for this?

Reply:
No, and this is kind of an unrelated thing that I encountered when trying a different fix. I might take this out of this PR.